lib/smart_city/event/dataset_update/technical.ex

```diff
@@ -0,0 +1,86 @@
+defmodule SmartCity.Event.DatasetUpdate.Technical do
+  @moduledoc """
+  A struct defining technical metadata on a dataset update event message.
+  """
+  alias SmartCity.Helpers
+
+  @type not_required(type) :: type | nil
+
+  @type t() :: %SmartCity.Event.DatasetUpdate.Technical{
+          dataName: String.t(),
+          orgName: String.t(),
+          systemName: String.t(),
```

Comment:
Could we get rid of systemName in this move?
(And create a systemName function or something that returns orgName__dataName?